### PR TITLE
Analysis refactoring

### PR DESCRIPTION
**Pull request by ibbem**
Refactoring of the analysis machinery. These changes where motivated by the changes in the `thesis_lb` branch and where originally discussed in #63. Note that there are some unrelated changes because of this.

### CODE REVIEW

**Comment by ibbem** [comment]
Note that 8370477ab42aed88f8e615432f3f09bacf05ae84 is an artifact of #63 where it was initially introduced by TheBormann.

File: src/main/java/org/variantsync/diffdetective/metadata/Metadata.java [hunk 1]
```diff
@@ -37,6 +40,18 @@ default void append(T other) {
         semigroup().appendToFirst(Cast.unchecked(this), other);
     }
 
+    static <T> T mergeEqual(T a, T b) {
```
Comment (pmbittner) [inline src/main/java/org/variantsync/diffdetective/metadata/Metadata.java:43]: Document behaviour and intention?

**Review by pmbittner — APPROVED** [review]
Only one comment and afterwards we can merge.

**Comment by ibbem** [comment]
Added documentation for `Metadata.mergeEqual`, `AnalysisResult.ResultKey`, `AnalysisResult.get` and `AnalysisResult.append`.

**Pull request merged**